frontend/public/components/modals/delete-modal.tsx

```diff
@@ -15,17 +15,19 @@ import { YellowExclamationTriangleIcon } from '@console/shared';
 import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager/src/models';
 import { findOwner } from '../../module/k8s/managed-by';
 import { ResourceLink } from '../utils/resource-link';
+import { cleanUpWorkload } from '@console/topology/src/utils';
```

Comment:
We should try to avoid adding (new) cycle dependencies.

Topology and dev console uses `@console/internal` (public folder) a lot, but the internal package should not import from topology - when possible.

In this case, I would recommend moving the `cleanUpWorkload` function into `@console/internal` as well, or `@console/shared`.


**_But_** even better: Instead of changing `CommonActionFactory` and add knative specific features into our shared code, it will be much cleaner if you would override the Delete action. Then this (multiple) cleanup functions would already knows the type and its not required to pass "isKnative" to a shared component.